### PR TITLE
Issue #559: Add --fable opt-in to run-spec.sh

### LINKED ISSUE #559: run-spec.sh に --fable opt-in 追加（コスト/retention 警告付き、ZDR graceful degrade）
## 背景

Claude Fable 5 は long-horizon 推論・設計品質で最も効く一方、$10/$50 per MTok（Opus 4.8 の 2 倍）、サブスクでは 2026-06-22 以降 usage credits 必須、ZDR 組織では利用不可（30 日 retention 必須）という採用制約を持つ。したがってデフォルト化は不可で、**選択的 opt-in** が妥当。

spec フェーズは設計品質が critical でエラーが全後続フェーズへ伝播するため、Fable 5 を opt-in する単一最高レバレッジの箇所。`run-spec.sh` は現状 `--opus`（xhigh/max）の opt-in を持つ。

詳細: `docs/reports/claude-fable-5-impact-strategy.md` §5.2、§3.3。

## 目的

`run-spec.sh` に `--fable` opt-in を追加し、L/XL spec で Fable 5 を選択可能にする。コスト/retention の警告を出し、ZDR 環境では graceful degrade する。デフォルトは変更しない。

## 提案

- Fable 5 のモデル文字列は `docs/tech.md` §C4 で `claude-fable-5`（明示的フル ID 必須、`opus` エイリアス経由は不可）と確認済み。`fable` 短縮エイリアスは存在しない。
- `run-spec.sh` に `--fable` オプションを追加（`MODEL` を `claude-fable-5` に、`EFFORT` は `high` を既定とし `--max` で `max`）。
- 実行開始バナーに警告を出力: コスト（$10/$50）、サブスク credit ゲート（2026-06-22 以降）、retention 要件。
- ZDR / Fable 5 不可環境での graceful degrade: 利用不可を検知できる場合は Opus 4.8（`--opus` 相当）へフォールバックし、その旨を stdout に明示。検知できない場合でも警告のみで強制終了はしない（Non-Goal 準拠）。
- `docs/tech.md` model-effort-matrix（§C4）への追記と整合させる。

### Pre-merge（自動検証）

- [x]  `run-spec.sh` が Fable 5 の正式モデル文字列 `claude-fable-5` を使用している
- [x]  `run-spec.sh` が `--fable` オプションを解釈する
- [x]  `--fable` 使用時に retention 警告が出力される
- [x]  `--fable` 使用時に credit 警告が出力される
- [x]  `run-spec.sh` が構文エラーなし
- [x]  Usage 文字列に `--fable` が追記されている
- [x]  既存の `--opus` / `--max` / デフォルト経路が回帰していないこと（bats テスト green）
- [x]  `docs/tech.md` model-effort-matrix §C4 に `--fable` opt-in の行が追記されている

### Post-merge

- [x] `--fable` 指定の `run-spec.sh` が Fable 5 で spec を生成できること（手動 1 回） 
- [x] ZDR / 不可環境で graceful degrade（フォールバックまたは警告継続）すること（手動または擬似環境） 

---

メタ（レポート §8）: Priority=high / Size=M / Phase=spec

## Auto-Resolved Ambiguity Points

1. **Fable 5 モデル文字列の確認・記録 AC を機械検証化**（影響：AC1 本文変更）
   - 選択: `file_contains "scripts/run-spec.sh" "claude-fable-5"` で代替
   - 根拠: `docs/tech.md` §C4 が `claude-fable-5` を既に文書化済み（「`opus` エイリアス経由では不可、明示的モデル文字列必須」）。実装が正式モデル文字列を使用しているかを機械的に検証できる。
   - 候補: rubric による記録確認 / file_contains による実装確認 → 後者を選択（確定的・低リスク）

2. **grep 引数逆順バグの修正**（影響：AC2, AC3, AC5 の verify command）
   - 選択: `grep "PATTERN" "PATH"` の正しい順序に修正
   - 根拠: `verify-executor.md` 翻訳テーブルの `grep "pattern" "path"` 仕様に合わせる。逆順のままでは全件 FAIL となる。
   - 候補: 修正 / そのまま → 修正を選択（正確性優先）

3. **Usage 検証の強化**（影響：AC5 の verify command）
   - 選択: `grep "scripts/run-spec.sh" "Usage"` → `file_contains "scripts/run-spec.sh" "[--fable]"` に変更
   - 根拠: `grep "Usage"` は既存ファイルにも "Usage:" が存在するため常に PASS となる。現在の usage 文字列が `[--opus]` `[--max]` 形式を採用しているため `[--fable]` が正確な検証対象。
   - 候補: 既存パターン維持 / より強い検証 → 強い検証を選択

4. **bats regression テスト verify command 追加**（影響：AC6 に verify command 追加）
   - 選択: `<!-- verify: command "bats tests/run-spec.bats" -->`
   - 根拠: `tests/run-spec.bats` が存在し、既存の `--opus` / `--max` / デフォルト経路をテストしている。
   - 候補: verify command なし / command "bats tests/run-spec.bats" → 後者を選択

5. **docs/tech.md 更新 AC の追加**（影響：pre-merge AC に1件追加）
   - 選択: `<!-- verify: grep "fable" "docs/tech.md" -->` で追加
   - 根拠: 提案セクションに「docs/tech.md model-effort-matrix（§C4）への追記と整合」が明記されているが AC に不在。tech.md に SSoT note（run-*.sh 変更時は必ず更新）あり。
   - 候補: AC 追加なし / AC 追加 → AC 追加を選択（要件の完全性）

6. **post-merge AC に verify-type タグ付与**（影響：post-merge 2件にタグ追加）
   - 選択: 両条件に `<!-- verify-type: manual -->`
   - 根拠: 実際の Fable 5 環境および ZDR 環境での動作確認は自動化不可能。

### PR DESCRIPTION
**Pull request by saito**
## Summary

- `scripts/run-spec.sh` に `--fable` オプションを追加（`MODEL=claude-fable-5`、`EFFORT=high` デフォルト、`--max` で `max`）
- `--fable` 使用時にコスト・サブスク credit ゲート・30日 retention 要件の警告を出力
- `tests/run-spec.bats` に `--fable` 動作の bats テスト 5 件追加
- `docs/tech.md` および `docs/ja/tech.md` の model-effort-matrix に Fable 5 情報を追記

## Verification (pre-merge)

- `run-spec.sh` が Fable 5 の正式モデル文字列 `claude-fable-5` を使用している
- `run-spec.sh` が `--fable` オプションを解釈する
- `--fable` 使用時に retention 警告が出力される
- `--fable` 使用時に credit 警告が出力される
- `run-spec.sh` が構文エラーなし
- Usage 文字列に `--fable` が追記されている
- 既存の `--opus` / `--max` / デフォルト経路が回帰していないこと（bats テスト green）
- `docs/tech.md` model-effort-matrix §C4 に `--fable` opt-in の行が追記されている

## Verification (post-merge)

- `--fable` 指定の `run-spec.sh` が Fable 5 で spec を生成できること（手動 1 回）
- ZDR / 不可環境で graceful degrade（フォールバックまたは警告継続）すること（手動または擬似環境）

closes #559

### CODE REVIEW

File: tests/run-spec.bats [hunk 1]
```diff
@@ -318,3 +318,34 @@ MOCK
...
+    run bash "$SCRIPT" 123 --fable
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"retention"* ]]
+}
+
+@test "success: --fable outputs credit warning" {
+    run bash "$SCRIPT" 123 --fable
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"credit"* ]]
+}
```
Comment (saito) [inline tests/run-spec.bats:351]: **[Edge Case/Robustness]** コスト警告（`$10/$50 per MTok`）のテストが不在。`retention` と `credit` は bats テストで検証済みですが、3行目の警告（コスト行）は同一 `if` ブロック内にあるにも関わらず明示的なテストがありません。

Severity: CONSIDER

推奨修正:
```bats
@test "success: --fable outputs cost warning" {
    run bash "$SCRIPT" 123 --fable
    [ "$status" -eq 0 ]
    [[ "$output" == *"cost"* ]]
}
```

File: scripts/run-spec.sh [hunk 2]
```diff
@@ -1,9 +1,9 @@
 #!/bin/bash
 # run-spec.sh - Autonomous /spec execution with Sonnet model
-# Usage: run-spec.sh <issue-number> [--opus] [--max]
+# Usage: run-spec.sh <issue-number> [--opus] [--fable] [--max]
```
Comment (saito) [inline scripts/run-spec.sh:3]: **[Documentation Consistency]** 直上のコメント（行2）が `# run-spec.sh - Autonomous /spec execution with Sonnet model` のままで、Opus / Fable 5 も使用できる現状と乖離しています。

Severity: CONSIDER

推奨修正: `# run-spec.sh - Autonomous /spec execution (Sonnet by default; --opus for Opus, --fable for Fable 5)`

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| 条件 | 結果 | 備考 |
|------|------|------|
| run-spec.sh が claude-fable-5 を使用 | PASS | — |
| run-spec.sh が --fable を解釈 | PASS | 4箇所（行3, 6, 20, 31） |
| --fable 使用時に retention 警告が出力 | PASS | 行66 |
| --fable 使用時に credit 警告が出力 | PASS | 行65 |
| run-spec.sh が構文エラーなし | UNCERTAIN | safe モード; CI "Run bats tests" (SUCCESS) が間接カバレッジを提供 |
| Usage 文字列に --fable が追記 | PASS | 3箇所（行3, 6, 31） |
| bats テスト green（回帰なし） | PASS | CI ref fallback: "Run bats tests" (SUCCESS) |
| docs/tech.md に --fable 行が追記 | PASS | 行76, 95 |

## CI Status

| ジョブ名 | ステータス | 備考 |
|---------|-----------|------|
| DCO | SUCCESS | — |
| Run bats tests | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## Code Review

### General Comments（path/line 特定不可の指摘）

なし。

### Issue Count Summary

- ラインコメントとして投稿: 2 件（MUST: 0, SHOULD: 0, CONSIDER: 2）
- General Comments（本文内）: 0 件

---
Next: run `/merge 570`.

**Comment by saito** [comment]
## Review Response Summary


### Claude Review Response

| アスペクト | ファイル | 行 | 指摘 | 対応 | 理由 |
|-----------|---------|-----|------|------|------|
| Edge Case/Robustness | tests/run-spec.bats | 351 | コスト警告テストが不在 | スキップ | CONSIDER レベル、3警告は同一 if ブロックで retention/credit テストが間接カバー |
| Documentation Consistency | scripts/run-spec.sh | 2 | コメントが "Sonnet model" のまま | スキップ | CONSIDER レベル、Usage 文字列は正確に更新済み |

### Lightweight Re-check Results

- テスト/検証: PASS（CI "Run bats tests" SUCCESS、"Validate skill syntax" SUCCESS、"Forbidden Expressions check" SUCCESS）
- 修正差分チェック: 対象なし（MUST 指摘なしのため修正作業なし）
- 新規指摘: なし
